### PR TITLE
cmake: fix up kokkos-3.0 update

### PR DESCRIPTION
**Pull request by junghans**
Fix up for #1870.

### CODE REVIEW

**Comment by junghans** [comment]
I tested all three path ways on Fedora 31. However it fails with Fedora's `kokkos-devel` due to the missing patch.

File: cmake/Modules/Packages/KOKKOS.cmake [hunk 1]
```diff
@@ -1,21 +1,20 @@
 if(PKG_KOKKOS)
-  find_package(Kokkos 3 QUIET)
-  if(Kokkos_FOUND)
-    set(DOWNLOAD_KOKKOS_DEFAULT OFF)
-  else()
-    set(DOWNLOAD_KOKKOS_DEFAULT ON)
-  endif()
-  option(DOWNLOAD_KOKKOS "Download the KOKKOS library instead of using an already installed one" ${DOWNLOAD_KOKKOS_DEFAULT})
+  option(EXTERNAL_KOKKOS "Build against external kokkos library")
+  option(DOWNLOAD_KOKKOS "Download the KOKKOS library instead of using an already installed one" OFF)
```
Comment (junghans) [inline cmake/Modules/Packages/KOKKOS.cmake]: This is kind of preview, how things will look without the bundled kokkos.

**Comment by stanmoore1** [comment]
@junghans here is the patch in case you need it: https://github.com/kokkos/kokkos/compare/master...stanmoore1:lammps

**Comment by junghans** [comment]
@stanmoore1 I am using https://github.com/kokkos/kokkos/compare/3.0.00...stanmoore1:lammps.diff as I want the patch against version 3.0.00. And I removed `cmake/2826.patch` as it was obviously wrong.

In CMake there are 3 options now, download, external and bundled kokkos. I left the download option as it was already merged in master.

**Comment by stanmoore1** [comment]
Ah, I misread the diff, I didn't see that the patch was removed.

**Pull request merged**